File: RansomNote.cpp
// Runtime: 17ms - Beats 87.33%
// Memory: 8.8MB - Beats 72.39%
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        for (int i = 0; i < ransomNote.length(); i++){
            bool flag = true;

            for (int j = 0; j < magazine.length(); j++){
                if (ransomNote[i] == magazine[j]){
                    magazine[j] = '.';
                    flag = false;
                    break;
                }
            }
            if (flag)
                return false;
        }
        return true;
    }
};

File: RansomNote.py
# Runtime: 49ms - Beats 95.4%
# Memory: 14.1MB - Beats 94.83%
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        for i in ransomNote:
            flag: bool = True

            for j in magazine:
                if i is j:
                    magazine = magazine.replace(j, '', 1)
                    flag = False
                    break
            
            if flag:
                return False
        return True
